Code/Network/RKManagedObjectRequestOperation.m

```diff
@@ -377,11 +377,28 @@ - (RKMappingResult *)refetchedMappingResult
         }
     } else {    
         for (NSString *destinationKeyPath in mappingInfo.relationshipMappingInfo) {
-            id relationshipValue = [object valueForKeyPath:destinationKeyPath];
+            id relationshipValue = nil;
+            // Objects in collection may have different types, so destination keypath may be not applicable to each of them
+            if([object conformsToProtocol:@protocol(NSFastEnumeration)]) {
+                NSMutableSet* results = [NSMutableSet set];
+                for (id item in object) {
+                    @try {
+                        id value = [item valueForKeyPath:destinationKeyPath];
+                        [results addObject:value];
+                    } @catch(NSException*) {
+                        continue;
+                    }
+                }
+                
+                relationshipValue = results;
+            } else {
+                relationshipValue = [object valueForKeyPath:destinationKeyPath];
+            }
+            
             NSArray *mappingInfos = (mappingInfo.relationshipMappingInfo)[destinationKeyPath];
             for (RKMappingInfo *relationshipMappingInfo in mappingInfos) {
                 NSUInteger index = [mappingInfos indexOfObject:relationshipMappingInfo];
-                id mappedObjectAtIndex = ([relationshipValue respondsToSelector:@selector(objectAtIndex:)]) ? [NSSet setWithObject:relationshipValue[index]] : relationshipValue;
```

Comment:
This fix provides proper processing of mapped objects' properties. Moreover, there is no need to wrap relationshipValue into NSSet